### PR TITLE
feat: Allow deleting Tracks

### LINKED ISSUE #911: Allow deleting a Track
### Feature Request

**Feature Description**

Allow deleting a Track

**Would you like to work on the issue?**

Yes

**Parent issue**

#870

### PR DESCRIPTION
**Pull request by srv-twry**
Fixes #911 

Changes: Add a delete icon
Delete the Track on the Server and the local database

Screenshots for the change:  
videotogif_2018 05 01_13 57 50

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Ask for confirmation before deleting

File: app/src/main/java/org/fossasia/openevent/app/core/track/list/viewholder/TracksViewHolder.java [hunk 1]
```diff
@@ -30,12 +31,19 @@ public void setEditAction(Pipe<Track> editAction) {
...
+
     public void bind(Track track) {
         trackId = track.getId();
         binding.setTrack(track);
         binding.actionChangeTrack.setOnClickListener(view -> {
             if (editAction != null) editAction.push(track);
         });
+        binding.actionDeleteTrack.setOnClickListener(view -> {
+            if (deleteAction != null) deleteAction.push(track);
+        });
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/track/list/viewholder/TracksViewHolder.java]: Move these listeners in constructors

**Comment by srv-twry** [comment]
videotogif_2018 05 01_14 37 26

File: app/src/main/java/org/fossasia/openevent/app/core/faq/list/FaqListFragment.java [hunk 2]
```diff
@@ -156,7 +156,7 @@ public void showDeleteDialog() {
         if (deleteDialog == null)
             deleteDialog = new AlertDialog.Builder(context)
                 .setTitle(R.string.delete)
-                .setMessage(R.string.delete_confirmation_message)
+                .setMessage(R.string.delete_confirmation_message_faq)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/faq/list/FaqListFragment.java]: Unneeded change

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/core/faq/list/FaqListFragment.java]: Change to template string

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -210,11 +210,13 @@
     <string name="delete_copyright">Delete Copyright</string>
     <string name="year_validation_error">Please Enter a Valid Year</string>
     <string name="delete_confirmation_message">Are you sure you want to delete the Question?</string>
+    <string name="delete_confirmation_message_track">Are you sure you want to delete the Track?</string>
```
Comment (iamareebjamal) [inline app/src/main/res/values/strings.xml]: Make thistemplate string?

Comment (srv-twry) [reply app/src/main/res/values/strings.xml]: Sure, so should i also use the created template string for the FAQ also ?

**Pull request merged**